src/Components/RecsListTable/RecsListTable.spec.ct.js

```diff
@@ -399,12 +404,17 @@ describe('successful non-empty recommendations list table', () => {
                 )
               );
           }
-
+          const filteringTags = filterData(DEFAULT_FILTERS).map((obj) => ({
+            ...obj,
+            tags: obj.tags.filter((string) =>
+              Object.keys(RULE_CATEGORIES).includes(string)
+            ),
+          }));
```

Comment:
Nice idea, but I see 2 issues:
1- If I am not mistaken, this code will modify the underlying objects in the data, thus it can affect the test afterwards. We need to create the modified data object outside the test, so all other tests can make use of it, or inside the test; but using a copy, so the data used by the remaining tests is unaltered
2- These code does not reflect exactly what the code does. Simply add "performace" to `"rule_id": "external.rules.rule_n_two|ERROR_KEY_N42"` and the test will fail. That is because you are comparing by an array of elements in the test, but in the code you use only the first one. The test should reflect what is the expected behavior.
You need to take the first element of each array to fix issue 2 and create a copy of the object to fix issue 1.

Reply:
Thanks @Fewwy for the clarification, I was wrong with my assumption on 1

Reply:
Thanks for your help :) tests are HAAARD!